webapp/sources/rudder/rudder-web/src/main/javascript/rudder/tree.js

```diff
@@ -634,13 +629,15 @@ function openTreeNodes(treeId, settingsStorageId, data){
 }
 
 /**
- * Function that fills the `allEvent` global variable with js-tree event handlers. By default, techniqueNode and directiveNode selectors are used as tree nodes.
+ * Function that fills the `allEvents` map with js-tree event handlers. By default, techniqueNode and directiveNode selectors are used as tree nodes.
  */
 function allEventsRegisterTree() {
   // Save node event handlers, in order to customize node by reassigning the click event handler
-  var allNodes = $('li.techniqueNode > a[id^="lift-event-js-"], li.techniqueNode > * > span[id^="lift-event-js-"], li.directiveNode > a[id^="lift-event-js-"], li.directiveNode > * > span[id^="lift-event-js-"]');
-  var events =  allNodes.map(function () {
-    return $._data(this, 'events')["click"]; // we need to intercept the click now, the object will change after the jstree init
-  }).get().map(function (x, i) { return [allNodes[i], x] });  // cannot use a node object as key, so keep a [node, events] array
-  allEvents.push(...events);
+  const allNodes = $('li.techniqueNode > a[id^="lift-event-js-"], li.techniqueNode > * > span[id^="lift-event-js-"], li.directiveNode > a[id^="lift-event-js-"], li.directiveNode > * > span[id^="lift-event-js-"]');
+  allNodes.each(function () {
+    const events = $._data(this, 'events')?.click;
+    if (events && events[0]) {
```

Comment:
Why checking first element of events ? is the goal to check that events is not empty ? If so why not "events.length !== 0" ?